thread/common/threadhelpers.cpp

```diff
@@ -70,17 +70,23 @@ omrthread_spinlock_acquire(omrthread_t self, omrthread_monitor_t monitor)
 	uintptr_t spinCount2Init = monitor->spinCount2;
 	uintptr_t spinCount1Init = monitor->spinCount1;
 
-	uintptr_t spinCount3 = spinCount3Init;
-	uintptr_t spinCount2 = spinCount2Init;
-
 #if defined(OMR_THR_SPIN_WAKE_CONTROL)
- 	if (monitor->spinThreads < lib->maxSpinThreads) {
- 		VM_AtomicSupport::add(&monitor->spinThreads, 1);
- 	} else {
- 		goto exit;
- 	}
+	BOOLEAN spinning = TRUE;
+	if (OMRTHREAD_IGNORE_SPIN_THREAD_BOUND != lib->maxSpinThreads) {
```

Comment:
What is the default for maxSpinThreads?

If maxSpinThreads == OMRTHREAD_IGNORE_SPIN_THREAD_BOUND then shouldn't the code still do the new behaviour?  If not what is the point of this new code?

Reply:
the purpose of `OMRTHREAD_IGNORE_SPIN_THREAD_BOUND != lib->maxSpinThreads` is to remove the bound on the number of spin threads, and allow unlimited spin threads. this takes us back to early 2017 or late 2016 when there was no bound on the number of spin threads. default on `maxSpinThreads` varies on the number of CPUs on a machine: https://github.com/eclipse/openj9/pull/553/commits/d4823cdb8f91e9219124b78115f2d91929d22c72.